docs/modules/ROOT/pages/explanations/decisions/deletion-protection.adoc

```diff
@@ -0,0 +1,129 @@
+There are multiple layers of deletion protection to consider:
+
+* Accidental deletion by an end-user -> for example deleting a claim without intending to
+* Accidental deletion on the backend -> for example a bug in the composition-function unintentionally removes something
+
+Currently, there's a deletion protection for PostgreSQL in place with a separate controller.
+However, this controller only protects against accidental deletion of the claim, nothing else.
+It also lacks support for other services and contains a few bugs which prevent it from cleaning up the resources after the grace period.
+
+=== Goals
+
+* Easy to use end-user deletion protection
+* Easy to extend and integrate backend deletion protection
+
+=== Non-Goals
+
+* N/A
+
+== Proposals
+The proposals are split between the end-user and backend deletion protection.
+
+== End-user Proposal 1: Protect claim deletion via webhooks
+The deletion of the claim is blocked via a webhook.
+There will be a new field in every claim that controls the deletion protection.
+For example `.spec.parameters.backup.deletionProtection`, if set to true, the webhook will not allow any deletion operations against the claim.
+To delete a claim, the field has to be explicitly set to a specific value (like `false`), and be applied with that value first.
+
+Advantages::
+* No actual deletion happens
+* No messing around with finalizers
+* No undelete process necessary, no deletion happened to begin with
+* Directly provides feedback to the user with a helpful message
+* Apart from the webhooks no separate controller is necessary for the implementation
+
+Disadvantages::
+* Breaks some k8s expected behavior
+* Changing the value in a GitOps environment, as the claim needs to be committed with the disabled deletion protection first
+* Not possible to implement a grace period, as no `deletionTimestamp` is set
+
+== End-user Proposal 2: Protect claims via finalizers and separate controller
+There's a controller which watches all claims and sets an additional finalizer on them.
+Additionally, all relevant objects within the service also get a finalizer, for example the namespace or pvcs.
+This finalizer stays on the objects as long as the deletion protection is active.
+It's controlled via the claim for example via `.spec.parameters.backup.deletionProtection`.
+
+Advantages::
+* A grace period can be implemented, after which the controller removes the finalizers
+* The existing controller can be extended with the features
+
+Disadvantages::
+* Each service will need its own configuration to specify which objects need finalizers
+* As Crossplane doesn't trigger any reconciles after the `deletionTimestamp` is set, a separate controller is necessary
+* The `deletionTimestamp` gets set on the objects, there's no way to undelete the object at this point. Any important data or manifest needs to be copied away
```

Comment:
I agree, when we stick with finalizers we can only protect something AFTER it's already doomed. The main reason I suggest to use webhooks.

This idea actually came to me when I did the cleanup of the sgpoolconfig objects. IF StackGres is running, it will perfectly ensure that no objects are deleted that are still being used.